### PR TITLE
Add mypy error codes to `type: ignore`s, remove unused ignores

### PR DESCRIPTION
**Pull request by AlexWaygood**
Accomplished with a little help from @Akuli's script in https://github.com/python/typeshed/pull/6379/commits/745da721810c469f3fe2ae39a8ddd84e1b773d07

This does the uncontroversial bits from https://github.com/python/typeshed/issues/7497 (adds mypy error codes where possible, removes the unused ignores, highlights which ignores are pyright-specific). We can carry on discussing in the issue thread whether we want to change the pyright-specific `type: ignore`s to use the new `pyright: ignore` comments instead.

### CODE REVIEW

**Comment by AlexWaygood** [comment]
@hauntsaninja, any idea which config options we're missing from `mypy_test.py` that means stubtest is picking up on mypy build errors [here](https://github.com/python/typeshed/runs/5583967229?check_suite_focus=true) that were missed by `mypy_test` [here](https://github.com/python/typeshed/runs/5583962106?check_suite_focus=true)?

**Review by srittau — APPROVED** [review]
Thanks and LGTM! Cc @Akuli

File: stdlib/@python2/smtpd.pyi [hunk 1]
```diff
@@ -10,7 +10,7 @@ class SMTPChannel(asynchat.async_chat):
     DATA: int
     def __init__(self, server: SMTPServer, conn: socket.socket, addr: Any, data_size_limit: int = ...) -> None: ...
     # base asynchat.async_chat.push() accepts bytes
```
Comment (JelleZijlstra) [inline stdlib/@python2/smtpd.pyi:12]: should remove this comment
```suggestion
```

File: stdlib/@python2/typing_extensions.pyi [hunk 2]
```diff
@@ -48,7 +48,7 @@ class _TypedDict(Mapping[str, object], metaclass=abc.ABCMeta):
     # can go through.
     def setdefault(self, k: NoReturn, default: object) -> object: ...
     # Mypy plugin hook for 'pop' expects that 'default' has a type variable type.
```
Comment (JelleZijlstra) [inline stdlib/@python2/typing_extensions.pyi:50]:
```suggestion
```

File: stubs/boto/boto/s3/connection.pyi [hunk 3]
```diff
@@ -123,4 +123,17 @@ class S3Connection(AWSAuthConnection):
...
+        key: str = ...,
+        headers: Any | None = ...,
+        data: str = ...,
+        query_args: Any | None = ...,
+        sender: Any | None = ...,
+        override_num_retries: Any | None = ...,
+        retry_handler: Any | None = ...,
+        *args,
+        **kwargs,
+    ): ...  # https://github.com/python/mypy/issues/1237
```
Comment (JelleZijlstra) [inline stubs/boto/boto/s3/connection.pyi:139]: This issue is closed, doesn't seem very useful to refer to it.

File: stubs/boto/boto/connection.pyi [hunk 4]
```diff
@@ -166,7 +166,9 @@ class AWSQueryConnection(AWSAuthConnection):
         provider: str = ...,
     ) -> None: ...
     def get_utf8_value(self, value): ...
-    def make_request(self, action, params: Any | None = ..., path: str = ..., verb: str = ..., *args, **kwargs): ...  # type: ignore # https://github.com/python/mypy/issues/1237
+    def make_request(  # type: ignore[override]
+        self, action, params: Any | None = ..., path: str = ..., verb: str = ..., *args, **kwargs
+    ): ...  # https://github.com/python/mypy/issues/1237
```
Comment (AlexWaygood) [inline stubs/boto/boto/connection.pyi:171]:
```suggestion
    ): ...
```

File: stubs/boto/boto/s3/connection.pyi (hunk 3 above)
Comment (AlexWaygood) [reply stubs/boto/boto/s3/connection.pyi:139]:
```suggestion
    ): ...
```

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**